multinet/db.py

```diff
@@ -387,7 +434,12 @@ def aql_query(workspace: str, query: str) -> Generator[Any, None, None]:
     """Perform an AQL query in the given workspace."""
     aql = get_workspace_db(workspace).aql
 
-    cursor = aql.execute(query)
+    try:
+        aql.validate(query)
+        cursor = aql.execute(query)
+    except (AQLQueryValidateError, AQLQueryExecuteError) as e:
+        raise MalformedRequestBody(str(e))
```

Comment:
This chunk should probably become a function (it's largely identical to a code hunk above).

Reply:
8d4355d